core/templates/services/prevent-page-unload-event.service.ts

```diff
@@ -25,8 +25,9 @@ import { WindowRef } from 'services/contextual/window-ref.service';
 })
 export class PreventPageUnloadEventService {
   private listenerActive: boolean;
-  validationCallback: Function;
-  _preventPageUnloadEventHandlerBind = undefined;
+  validationCallback?: (() => boolean);
+  _preventPageUnloadEventHandlerBind?: (
+    this: Window, ev: BeforeUnloadEvent) => void;
```

Comment:
we use `(!)` in places where we are sure it would never be `null` or `undefined`. That s not the case here. Like in the test if `removeListener()` is called before adding a listener, `_preventPageUnloadEventHandlerBind` will be `undefined`